configs/components/rubygem-nokogiri.rb

```diff
@@ -7,12 +7,15 @@
 
   gem_home = settings[:gem_home]
   pkg.environment "GEM_HOME", gem_home
-
-  # When cross compiling nokogiri native extensions on macOS 11/12 ARM, there is a 94M tmp
-  # directory that's not needed
-  if platform.is_macos? && platform.architecture == 'arm64'
-    install do
-      "rm -r #{gem_home}/gems/nokogiri-#{pkg.get_version}/ext/nokogiri/tmp"
+  if platform.is_macos?
+    pkg.environment "PKG_CONFIG_PATH", "#{settings[:libdir]}/pkgconfig"
+    pkg.install do
+      "#{settings[:gem_install]} nokogiri --platform=ruby -- --use-system-libraries --with-opt-dir=/opt/puppetlabs/puppet"
```

Comment:
One subtle issue, I think `gem install nokgiri` behaves differently than `gem install nokogiri.gem`. I think the former will download the gem from rubygems/artifactory, while the latter will see that it's a local gem on disk and should install that.

Reply:
I tried this, specifying `#{name}-#{pkg.get_version}.gem`, and got the same warning about libxml: `WARNING: Nokogiri was built against libxml version 2.9.13, but has dynamically loaded 2.12.6`.